components/pedigree/resources/src/main/resources/pedigree/detailsDialogueGroup.js

```diff
@@ -0,0 +1,540 @@
+                    }
+                    _this._crtFocus = addedDialogue;
+                    _this._focus(_this._crtFocus);
+                } else {
+                    if (_this._crtFocus) {
+                        _this._blur(_this._crtFocus);
+                        _this._crtFocus = null;
+                    }
+                }
+            });
+            // Observe clicks.
+            document.observe('click', function (event) {
+                var clickedDialogueHolder = event.findElement('td.dialogue-holder');
+                if (clickedDialogueHolder === _this._dialogueHolder) {
+                    var summaryItem = event.findElement('div.summary-item');
+                    if (_this._crtFocus) {
+                        if (_this._crtFocus !== summaryItem) {
+                            _this._blur(_this._crtFocus);
+                            _this._crtFocus = summaryItem;
+                            _this._focus(_this._crtFocus);
+                        }
+                    } else {
+                        _this._crtFocus = summaryItem;
+                        _this._focus(_this._crtFocus);
+                    }
+                } else {
+                    if (_this._crtFocus && (!event.target.hasClassName('term-status')
+                        || _this._dialogueHolder.up('table') !== clickedDialogueHolder.up('table'))) {
+
+                        _this._blur(_this._crtFocus);
+                        _this._crtFocus = null;
+                    }
+                }
+            });
+        },
+
+        /**
+         * Blurs a dialogue element.
+         *
+         * @param elem a dialogue element
+         * @private
+         */
+        _blur: function(elem) {
+            var elemID = elem.down('input').value;
+            var dialogue = this._dialogueMap[elemID];
+            if (dialogue) {
+                dialogue.blur();
+                Event.fire(this._qualifiersContainer, this._dataName + ':dialogue:blurred');
+            }
+        },
+
+        /**
+         * Focuses a dialogue element.
+         *
+         * @param elem a dialogue element
+         * @private
+         */
+        _focus: function(elem) {
+            var elemID = elem.down('input').value;
+            var dialogue = this._dialogueMap[elemID];
+            if (dialogue) {
+                dialogue.focus();
+                Event.fire(this._qualifiersContainer, this._dataName + ':dialogue:focused');
+            }
+        },
+
+        /**
+         * Attaches an observer to the _dialogueHolder, that monitors any 'keyup' events on any object with
+         * class name 'qualifier-notes'. Fires a custom events when notes are updated.
+         * Code adapted from https://stackoverflow.com/a/4220182
+         *
+         * @private
+         */
+        _attachKeyUpObserver: function() {
+            var _this = this;
+            var typingTimer;
+            var doneTypingInterval = 500;
```

Comment:
we may want to make the interval be part of the constructor parameters (`options`)

Reply:
Fixed.